pkg/controller/persistence/dgraph/service.go

```diff
@@ -93,5 +92,84 @@ func PersistServicesInteractionGraph(sourceService string, destinationServices [
 	}
 	_, err = MutateNode(Client, bytes)
 	return err
+}
+
+// PersistPodsInServices saves pods in Services object in the dgraph
+func PersistPodsInServices(svcToPod map[string][]string) error {
+	for svcXID, podXIDs := range svcToPod {
+		svcUID := GetUID(Client, svcXID, IsService)
+		if svcUID == "" {
+			continue
+		}
+
+		svcPods := []*Pod{}
+		for _, podXID := range podXIDs {
+			podUID := GetUID(Client, podXID, IsPod)
+			if podUID == "" {
+				log.Debugf("Pod uid is empty for pod xid: %s", podXID)
+				continue
+			}
+			pod := &Pod{
+				ID: ID{UID: podUID},
+			}
+			svcPods = append(svcPods, pod)
+		}
+
+		updatedService := Service{
+			ID:  ID{UID: svcUID},
+			Pod: svcPods,
+		}
+		bytes := utils.JsonMarshal(updatedService)
+		_, err := MutateNode(Client, bytes)
+		if err != nil {
+			log.Error(err)
+		}
+	}
+	return nil
```

Comment:
We are not returning `error` here. Either return the error or remove the return stmt.